tasks/setup_registry.yaml

```diff
@@ -0,0 +1,98 @@
+---
+# tasks file for configure-local-registry
+
+- name: Create registry directories
+  file:
+    path: /opt/registry/{{ item }}
+    state: directory
+  with_items:
+    - auth
+    - certs
+    - data
+
+- name: Create self signed certificate for local registry
+  shell: openssl req -newkey rsa:4096 -nodes -sha256 -keyout domain.key -x509 -days 365 -out domain.crt \
+          -subj "/C=US/ST=State/L=City/O=Company, Inc./OU=Department Name/CN={{ registry_host }}"
+  args:
+    chdir: /opt/registry/certs
+
+- name: Copy self-signed certificate
+  copy:
+    src: /opt/registry/certs/domain.crt
+    dest: /etc/pki/ca-trust/source/anchors/domain.crt
+    remote_src: yes
+    force: yes
+
+- name: Add the self-signed certificate to your list of trusted certificates
+  shell: update-ca-trust
+
+- name: Create the user and password for local registry
+  shell: htpasswd -bBc /opt/registry/auth/htpasswd admin admin
+
+- name: Generate local-registry service file
+  template:
+    src: ../templates/local-registry.service.j2
+    dest: /etc/systemd/system/local-registry.service
+    mode: 0655
+
+- name: Start local-registry
+  systemd:
+    name: local-registry
+    state: started
+    enabled: yes
+    daemon_reload: yes
+
+- name: Ensure registry pod is up
+  shell: podman ps | grep local-registry
+  register: pod_state
+  until: pod_state.stdout != ""
+  retries: 4
+  delay: 15
+
+- name: Configure firewalld service
+  block:
+    - name: Allow traffic at local registry port
+      firewalld:
+        port: 5000/tcp
+        permanent: yes
+        zone: "{{ item }}"
+        state: enabled
+      with_items:
+      - internal
+      - public
+
+    - name: Add http service
+      firewalld:
+        service: https
+        permanent: yes
+        state: enabled
+
+- name: Restarting regitry services
+  service:
+    name: "{{ item }}"
+    state: restarted
+  with_items:
+    - "{{ registry_services }}"
+
+- name: Get local registry pod
+  shell: curl -u admin:admin -k https://{{ local_registry }}/v2/_catalog | grep repositories
+  register: pod_state
+  until: pod_state.stdout != ""
+  retries: 3
+  delay: 30
+
+- name: Mirror the registry
+  when: pod_state.stdout != ""
+  block:
+  - name: Generate the base64-encoded user name and password or token for your mirror registry
+    shell: |
+      registry_token=`echo -n "admin:admin" | base64 -w0`
+      jq '.auths += {"{{ local_registry }}": {"auth": "'$registry_token'","email": "noemail@localhost"}}' < ~/pull-secret.json > ~/pull-secret-updated.json
+
+  - name: Mirror the registry
+    shell: oc adm -a ~/pull-secret-updated.json release mirror \
+      --from={{ release_image }} \
+      --to={{ local_registry }}/{{ local_repo }} \
+      --to-release-image={{ local_registry }}/{{ local_repo }}:{{ ocp_release }}
+    register: registry
```

Comment:
I would use a var here. Not everyone will have access to the internet if they are in a disconnected situation. I still like the idea and would keep it though!

Maybe add in the vars...

```
setup_registry:
  deploy: true
  autosync_registry: true
```

Then do something like ...

```
when: setup_registry.autosync_registry
```

The default is to set `setup_registry.autosync_registry` as false

Reply:
Done